### PR TITLE
Upgrade to Sparkle 2

### PR DESCRIPTION
**Pull request by ychin**
Upgrade from Sparkle 1 to Sparkle 2. Because Sparkle 2 now requires minimum macOS version to be 10.13, we can only use this for normal MacVim builds, but legacy binary (for 10.9-10.12) will need to use Sparkle 1 still. We now have separate Sparkle_1 / Sparkle_2.framework folders and symlink `Sparkle.framework` to `Sparkle_2.framework`. When doing legacy builds in CI, we change the symlink to point to `Sparkle_1.framework`, and set the corresponding compiler ifdef so the code will know what to do.

Also, add support for a new "prerelease" update channel (only available in Sparkle 2) which allows us to deploy pre-release / beta software to users who opt in (in the Advanced settings pane). This helps gather feedbacks during development before we push out full releases.

### CODE REVIEW

**Comment by ychin** [comment]
This relies on #1331 so needs that to be merged first.

**Comment by ychin** [comment]
Ping @ichizok if you are interested to take a look.

File: .github/workflows/ci-macvim.yaml [hunk 1]
```diff
@@ -40,23 +44,43 @@ jobs:
...
+      - name: Set up legacy build
+        if: matrix.legacy
+        run: |
+          # Set the correct build env vars to target the correct architectures and min OS targets.
+          echo "MACOSX_DEPLOYMENT_TARGET=$MACOSX_DEPLOYMENT_TARGET_LEGACY" >> $GITHUB_ENV
+          echo "MACVIM_ARCHS=$MACVIM_ARCHS_LEGACY" >> $GITHUB_ENV
+
+          # Use Sparkle 1 because Sparkle 2 requires newer OS version than our legacy build.
+          # Later, we pass the --enable-sparkle_1 flag to configure to set the corresponding ifdef.
+          rm src/MacVim/Sparkle.framework && ln -s Sparkle_1.framework src/MacVim/Sparkle.framework
```
Comment (ichizok) [inline .github/workflows/ci-macvim.yaml:82]: It can remove `rm ...` and change `ln -s ...` to `ln -fhs ...`.

**Comment by ychin** [comment]
Looks from CI that older versions of Xcode do not like a symlinked Sparkle.framework. Will need to do some investigation. This could be mildly annoying because someone on say macOS 10.13 may have problem building this unless they do some extra steps. Maybe instead of using a symlink I will just make `Sparkle.framework` the version 2 one, and just do the symlink trick only when building a legacy binary.

**Pull request merged**